sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -307,11 +307,12 @@ private[jdbc] class JDBCRDD(
       ""
     }
 
+    val myTopExpression: String = dialect.getTopExpression(limit) // SQL Server Limit alternative
```

Comment:
Yeah, feels like this belongs all in the MySQL dialect, one way or the other. Is it not just an alternative LIMIT clause?

Reply:
As I mentioned in the PR description, SQL Server does not support LIMIT clause.

Reply:
Of course, what i mean is, is "TOP n" just the 'limit' clause you need to generate for MySQL? then it doesn't need different handling, just needs to generate a different clause. But looks like it goes into the SELECT part, not at the end, OK

Reply:
MySQL supports LIMIT clause, I assume you meant MSSQL/SQL Server. 

Yes, you are right. Although TOP N has similar semantics as LIMIT, it is more of an expression (?) rather than a clause, sort of like DISTINCT. We need to insert TOP N before the list of columns in select, unfortunately, this is the syntax for it.

I came up with this approach of introducing another `getTopExpression` because I thought it would be the least intrusive one but I am open to alternative approaches, please let me know if you have any suggestions as to how I can refactor the code to support this.

Reply:
Oops yes I'm talking about what you're talking about, typo - MS SQL Server
I'm OK with it; the alternative is to somehow edit the SQL query down in the MS SQL dialect maybe, I haven't thought about it. It'd be nicer but not sure it's cleaner.

Reply:
Actually, there is already a precedent for adding something like this: `getTableSample` method.